### PR TITLE
Don't export HttpException

### PR DESCRIPTION
**Pull request by tvh**
This simplifies the error type. This allows for `http-client` to be swapped out (#51).

### CODE REVIEW

**Comment by alpmestan** [comment]
LGTM. @jkarni ?

File: servant-client/src/Servant/Common/Req.hs [hunk 1]
```diff
@@ -148,11 +148,11 @@ performRequest reqMethod req isWantedStatus reqHost = do
...
 
   eResponse <- liftIO $ __withGlobalManager $ \ manager ->
-    catchHttpException $
-    Client.httpLbs request manager
+    catchConnectionError $
+      Client.httpLbs request manager
   case eResponse of
     Left err ->
-      left $ ConnectionError err
+      left $ ConnectionError (show err)
```
Comment (christian-marie) [inline servant-client/src/Servant/Common/Req.hs]: Do we have to lose the underlying exception here? Can we just pass this thing along as a SomeException existential? Then the user can recover the exception, or just throw/show it.

**Comment by jkarni** [comment]
Hm, :+1: on moving away from exposing `http-client` internals, but I think `String` is not informative enough. Should we have our own `HttpException` datatype that's basically the same?

**Comment by christian-marie** [comment]
That might fix the things. I'm going to make a PR for this PR so we can merge whilst we merge.

**Comment by christian-marie** [comment]
This is my meta-PR: https://github.com/haskell-servant/servant/pull/118

**Comment by christian-marie** [comment]
Any thoughts on using SomeException?

**Comment by christian-marie** [comment]
Back to you @tvh

**Comment by tvh** [comment]
I only had a preliminary look at the errors returned by `XMLHttpRequest`. It seems to me though that there is a lot less information to be had there. I am all for `SomeException` here. If you want to handle the error, then you are free to do so, without losing information.

**Comment by christian-marie** [comment]
I think this is ready for merge.

**Comment by alpmestan** [comment]
LGTM. @jkarni ?

**Comment by jkarni** [comment]
Sure. I still fear we have less information than before, but if everyone thinks this is fine I'm good with it.

**Pull request merged**